rust/catalyst-voting/src/crypto/hash.rs

```diff
@@ -0,0 +1,50 @@
+//! Blake2b-256 hash implementation.
+
+use curve25519_dalek::digest::{
+    consts::U64, typenum::Unsigned, FixedOutput, HashMarker, Output, OutputSizeUser, Update,
+};
+
+/// Blake2b-512 hasher instance.
+pub struct Blake2b512Hasher(blake2b_simd::State);
```

Comment:
how does this affect the 1:1 mapping? I don't recall blake2b-512 being used.

Reply:
Here I've implemented a `digest::Digest` trait for `Blake2b512Hasher`.
And the `Blake2b512Hasher` is used here https://github.com/input-output-hk/catalyst-libs/blob/9fd9e0c144e1984e8096ae8d4e57f14e5fc7b192/rust/catalyst-voting/src/crypto/zk_dl_equality.rs#L53.
It allows me to build a Scalar value from hash using the interface provided by `curve25519_dalek` crate.
https://github.com/input-output-hk/catalyst-libs/blob/9fd9e0c144e1984e8096ae8d4e57f14e5fc7b192/rust/catalyst-voting/src/crypto/group/ristretto255.rs#L83

Reply:
can you run **_blake2b simd library_** through cargo audit, I recall there being a vulnerability. 

Just double check to make sure it can't affect us in anyway.